src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -271,6 +271,16 @@ struct AccessControlEntryCodec
         return CHIP_NO_ERROR;
     }
 
+    bool MatchesFabricIndex(FabricIndex fabricIndex) const
```

Comment:
Oof.

I wonder whether we should have MatchesFabricIndex on _everything_, and just have it always return true on the things that are not fabric-scoped.  The compiler would probably optimize that away pretty well...

@erjiaqing was there a reason not to do that?

Reply:
No special reasons.
However, I would suggest the approach in https://github.com/project-chip/connectedhomeip/pull/13902, then we can have the same API as https://github.com/project-chip/connectedhomeip/pull/13907

We do have the cases for getting the actual fabric index, in that case, I would suggest the getter solution.

Reply:
Another thought for 13902 is we should avoid putting logics in cluster object header.

My previous solution is not clean enough.

Reply:
The question is this: what could we have done so that code that was missing this required fabric-matching bit failed to compile?  The approach in #13902 _requires_ that it compile, because you can't hand out a fabric index for things that are not fabric-scoped, so really have to stick to asking the things that _are_ fabric-scoped.

Maybe the answer is that the check for being fabric-scoped should be a constexpr static function instead of a method-detect?  And require _that_ to be implemented for all structs?

Reply:
Thanks, I captured this discussion in issue #14430 and assigned it to @erjiaqing